### DISCUSSION TITLE
JSONPath/XPath access to Environment Variables

### DISCUSSION THREAD

**Discussion by js-mode** [Kong/insomnia#3731] [Ideas]
[Feature Request] 

Would like to ability to use specific workspace environment variables in the JSONPath queries. 

Currently we querying API with specific order numbers and am getting back very large JSON response for customer.  Would like to filter the JSONPath filter using the variable that is set on the GET API request. 

This would also be valuable for those that are constantly running the same JSONPath queries for that specific environment - for example if you are always running `$..[?(@.customer == '50')]` to filter out the information for just that environment (per call) so that you can have set environment variables you can use for JSON responses that are returned on an environment basis. 

The data may change, which is why it would be nice to parse out specific environment variables that you can throw into the JSONPath query section.  The history is very useful, but if the query is long enough, it is a lot easier to use specific variables (such as CustomerID, FQDNs, etc) to add into the JSONPath query.

-->

- Insomnia Version: Version 6.6.2 (6.6.2.2932)
- Operating System: macOSMojave version 10.14.16

**Comment by welcome** [comment]
👋 Thanks for opening your first issue! If you're reporting a 🐞 bug, please make sure
you include steps to reproduce it. If you're requesting a feature 🎁, please provide real
use cases that would benefit. 👪

To help make this a smooth process, please be sure you have first read the
[contributing guidelines](https://github.com/getinsomnia/insomnia/blob/master/CONTRIBUTING.md).

**Comment by js-mode** [comment]
I was looking through the codebase to see if I could figure out how to implement this. Not sure what the level of effort that is involved, but I think maybe the easiest way to approach this would be to include it in the existing insomnia-plugin-jsonpath and somehow tie it in to the environment variable lookup function that already exists in the main address bar. 

If that proves to be difficult, then maybe it is possible to add a separate Sub Environment that is dedicated to JSONPath filter/query variables that can be shared and reused amongst teams?

**Comment by gschier** [comment]
I like the idea of supporting environment variables in as many places as possible. Using them in JSONPath for the response makes sense I think.

There are three places JSONPath filters can be used, however, and one of them would be much more difficult to enable variable support.

1. The filter input box beneath the response
2. The Response template tag used for chaining requests (difficult)
3. In the JSONPath template tag (difficult)

The implementation for No. 1 would be fairly trivial and would probably cover most use cases. Is this where you are looking to use it? If so, I could hook this up in just a few minutes as we just need to render the filter string before passing it into the filter library.

**Comment by js-mode** [comment]
Hi! that is awesome.  No 1 is exactly what I was looking for and would be more than sufficient!

I also really like how the **filter input box** has the previous 10 filters that was used for that specific response and is very handy. 

Want to make sure that this would still work as it does now and wouldn't show the Environment Variable name, but the actual value. 

If there is something I can do to help test it out or whatever I can do to help, please let me know.

**Comment by js-mode** [comment]
@gschier -- would really liked to get an update on this feature. would be really helpful when I need to pass in Environment Variables for the response filters.

**Comment by nikcorg** [comment]
@gschier sorry for reviving a nearly 4 year old discussion, but did this feature ever make any progress? I would absolutely love this feature.
